lint/dead_code.py

```diff
@@ -0,0 +1,51 @@
+import ast
+
+import gamla
+import toolz
+from toolz import curried
+from toolz.curried import operator
+
+detect = gamla.compose_left(
+    ast.walk,
+    gamla.bifurcate(
+        gamla.compose_left(
+            curried.filter(
+                gamla.anyjuxt(
+                    gamla.is_instance(ast.AsyncFunctionDef),
+                    gamla.is_instance(ast.FunctionDef),
+                )
+            ),
+            curried.map(lambda function: function.name),
+        ),
+        gamla.compose_left(
+            curried.filter(gamla.is_instance(ast.Name)),
+            curried.map(lambda name: name.id),
+        ),
+        gamla.compose_left(
+            curried.filter(gamla.is_instance(ast.Attribute)),
+            curried.map(lambda attribute: attribute.attr),
+        ),
+    ),
+    toolz.concat,
+    curried.filter(lambda name: name.startswith("_")),
+    curried.remove(
+        operator.contains(
```

Comment:
Can't it be generalised to  all double underscore?

Reply:
You're on to me... I kinda want to discourage the usage of `__x__` :)